todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksActivity.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016, The Android Open Source Project
+ * Copyright (c) 2017-2018 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.architecture.blueprints.todoapp.tasks
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.design.widget.NavigationView
+import android.support.v4.view.GravityCompat
+import android.support.v4.widget.DrawerLayout
+import android.support.v7.app.ActionBar
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.view.MenuItem
+import com.example.android.architecture.blueprints.todoapp.R
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsActivity
+import com.example.android.architecture.blueprints.todoapp.util.ActivityUtils
+
+class TasksActivity : AppCompatActivity() {
+
+    private var mDrawerLayout: DrawerLayout? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.tasks_act)
+
+        // Set up the toolbar.
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolbar)
+        val ab = supportActionBar
+        supportActionBar?.apply {
+            setHomeAsUpIndicator(R.drawable.ic_menu)
+            setDisplayHomeAsUpEnabled(true)
+        }
+
+        // Set up the navigation drawer.
+        mDrawerLayout = findViewById(R.id.drawer_layout)
+        mDrawerLayout!!.setStatusBarBackground(R.color.colorPrimaryDark)
+        val navigationView = findViewById<NavigationView>(R.id.nav_view)
+        if (navigationView != null) {
+            setupDrawerContent(navigationView)
+        }
+
+        var tasksFragment: TasksFragment? = supportFragmentManager.findFragmentById(R.id.contentFrame) as TasksFragment?
+        if (tasksFragment == null) {
+            // Create the fragment
+            tasksFragment = TasksFragment.newInstance()
```

Comment:
the elvis operator would come in handy here :)

Reply:
can you tell this isn't really code I wrote? :D